### PR TITLE
Fix flutter_tools stuck when using custom LLDB prompt

### PR DESCRIPTION
**Pull request by LinXunFeng**
The problem only occurs on iOS devices

## Custom LLDB prompt
Open up the `~/.lldbinit`, type the following:

```shell
settings set prompt "(mylldb)"
```

## Problem

After custom LLDB prompt, `flutter run` will stuck on `The Dart VM service
is listening`.

```shell
flutter run -d [deviceid] -v
```

```shell
[        ] -------------------------
[ +306 ms] (mylldb)command source -s 0
'/tmp/07FCA604-600A-4CF4-A7FC-328809D4CFA3/fruitstrap-lldb-prep-cmds-00008120_0019784E2690C01E'
[        ] Executing commands in
'/tmp/07FCA604-600A-4CF4-A7FC-328809D4CFA3/fruitstrap-lldb-prep-cmds-00008120_0019784E2690C01E'.
[        ] (mylldb)    platform select remote-'ios' --sysroot
'/Users/lxf/Library/Developer/Xcode/iOS DeviceSupport/16.1 (20B82) arm64e/Symbols'
[        ]   Platform: remote-ios
[        ]  Connected: no
[        ]   SDK Path: "/Users/lxf/Library/Developer/Xcode/iOS DeviceSupport/16.1 (20B82)
arm64e/Symbols"
[        ] (mylldb)    target create
"/Users/lxf/Desktop/LXF/Test/abc/flutter_application_1/build/ios/iphoneos/Runner.app"
[+3552 ms] Current executable set to
'/Users/lxf/Desktop/LXF/Test/abc/flutter_application_1/build/ios/iphoneos/Runner.app' (arm64).
[        ] (mylldb)    script
fruitstrap_device_app="/private/var/containers/Bundle/Application/B0C8633C-5292-44FB-9523-1A3DEAA1A1
55/Runner.app"
[ +137 ms] (mylldb)    script fruitstrap_connect_url="connect://127.0.0.1:65264"
[        ] (mylldb)    script fruitstrap_output_path=""
[        ] (mylldb)    script fruitstrap_error_path=""
[        ] (mylldb)    target modules search-paths add /usr "/Users/lxf/Library/Developer/Xcode/iOS
DeviceSupport/16.1 (20B82) arm64e/Symbols/usr" /System "/Users/lxf/Library/Developer/Xcode/iOS
DeviceSupport/16.1 (20B82) arm64e/Symbols/System"
"/private/var/containers/Bundle/Application/B0C8633C-5292-44FB-9523-1A3DEAA1A155"
"/Users/lxf/Desktop/LXF/Test/abc/flutter_application_1/build/ios/iphoneos"
"/var/containers/Bundle/Application/B0C8633C-5292-44FB-9523-1A3DEAA1A155"
"/Users/lxf/Desktop/LXF/Test/abc/flutter_application_1/build/ios/iphoneos" /Developer
"/Users/lxf/Library/Developer/Xcode/iOS DeviceSupport/16.1 (20B82) arm64e/Symbols/Developer"
[  +13 ms] (mylldb)    command script import
"/tmp/07FCA604-600A-4CF4-A7FC-328809D4CFA3/fruitstrap_00008120_0019784E2690C01E.py"
[   +3 ms] (mylldb)    command script add -f fruitstrap_00008120_0019784E2690C01E.connect_command
connect
[        ] (mylldb)    command script add -s asynchronous -f
fruitstrap_00008120_0019784E2690C01E.run_command run
[        ] (mylldb)    command script add -s asynchronous -f
fruitstrap_00008120_0019784E2690C01E.autoexit_command autoexit
[        ] (mylldb)    command script add -s asynchronous -f
fruitstrap_00008120_0019784E2690C01E.safequit_command safequit
[        ] (mylldb)    connect
[  +36 ms] (mylldb)    run
[ +132 ms] success
[ +732 ms] (mylldb)2023-01-29 16:06:36.096176+0800 Runner[1379:446804] flutter: The Dart VM service
is listening on http://127.0.0.1:64244/9ISp5Qd_gkc=/

```

## Finally

I solved this problem by fetching lldb prompt dynamically.

```shell
[        ] -------------------------
[ +314 ms] (mylldb)command source -s 0
'/tmp/01D42AB4-E8DB-445F-8722-21CA5FE14647/fruitstrap-lldb-prep-cmds-00008120_0019784E2690C01E'
[        ] Executing commands in
'/tmp/01D42AB4-E8DB-445F-8722-21CA5FE14647/fruitstrap-lldb-prep-cmds-00008120_0019784E2690C01E'.
[        ] (mylldb)    platform select remote-'ios' --sysroot
'/Users/lxf/Library/Developer/Xcode/iOS DeviceSupport/16.1 (20B82) arm64e/Symbols'
[        ]   Platform: remote-ios
[        ]  Connected: no
[        ]   SDK Path: "/Users/lxf/Library/Developer/Xcode/iOS DeviceSupport/16.1 (20B82)
arm64e/Symbols"
[        ] (mylldb)    target create
"/Users/lxf/Desktop/LXF/Test/abc/flutter_application_1/build/ios/iphoneos/Runner.app"
[+3482 ms] Current executable set to
'/Users/lxf/Desktop/LXF/Test/abc/flutter_application_1/build/ios/iphoneos/Runner.app' (arm64).
[        ] (mylldb)    script
fruitstrap_device_app="/private/var/containers/Bundle/Application/D17A0F9D-F5B3-4355-96F0-EA4D3F6A13
12/Runner.app"
[ +133 ms] (mylldb)    script fruitstrap_connect_url="connect://127.0.0.1:49429"
[        ] (mylldb)    script fruitstrap_output_path=""
[        ] (mylldb)    script fruitstrap_error_path=""
[        ] (mylldb)    target modules search-paths add /usr "/Users/lxf/Library/Developer/Xcode/iOS
DeviceSupport/16.1 (20B82) arm64e/Symbols/usr" /System "/Users/lxf/Library/Developer/Xcode/iOS
DeviceSupport/16.1 (20B82) arm64e/Symbols/System"
"/private/var/containers/Bundle/Application/D17A0F9D-F5B3-4355-96F0-EA4D3F6A1312"
"/Users/lxf/Desktop/LXF/Test/abc/flutter_application_1/build/ios/iphoneos"
"/var/containers/Bundle/Application/D17A0F9D-F5B3-4355-96F0-EA4D3F6A1312"
"/Users/lxf/Desktop/LXF/Test/abc/flutter_application_1/build/ios/iphoneos" /Developer
"/Users/lxf/Library/Developer/Xcode/iOS DeviceSupport/16.1 (20B82) arm64e/Symbols/Developer"
[  +14 ms] (mylldb)    command script import
"/tmp/01D42AB4-E8DB-445F-8722-21CA5FE14647/fruitstrap_00008120_0019784E2690C01E.py"
[   +3 ms] (mylldb)    command script add -f fruitstrap_00008120_0019784E2690C01E.connect_command
connect
[        ] (mylldb)    command script add -s asynchronous -f
fruitstrap_00008120_0019784E2690C01E.run_command run
[        ] (mylldb)    command script add -s asynchronous -f
fruitstrap_00008120_0019784E2690C01E.autoexit_command autoexit
[        ] (mylldb)    command script add -s asynchronous -f
fruitstrap_00008120_0019784E2690C01E.safequit_command safequit
[        ] (mylldb)    connect
[  +31 ms] (mylldb)    run
[ +126 ms] success
[        ] Application launched on the device. Waiting for Dart VM Service url.
[ +773 ms] Observatory URL on device: http://127.0.0.1:64846/ImlqeV22izU=/
[   +7 ms] Attempting to forward device port 64846 to host port 49439
[        ] executing: /Users/lxf/Desktop/gitHub/flutter/bin/cache/artifacts/usbmuxd/iproxy
49439:64846 --udid 00008120-0019784E2690C01E --debug
[+1015 ms] Forwarded port ForwardedPort HOST:49439 to DEVICE:64846
[   +2 ms] Forwarded host port 49439 to device port 64846 for Observatory
[  +12 ms] Installing and launching... (completed in 8.7s)
[   +4 ms] Caching compiled dill
[  +55 ms] Connecting to service protocol: http://127.0.0.1:49439/ImlqeV22izU=/
[ +110 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM
service at http://127.0.0.1:49439/ImlqeV22izU=/.
[  +49 ms] DDS is listening at http://127.0.0.1:49444/Iv9OxkIC-sQ=/.
[  +22 ms] Successfully connected to service protocol: http://127.0.0.1:49439/ImlqeV22izU=/
[  +16 ms] DevFS: Creating new filesystem on the device (null)
[  +13 ms] DevFS: Created new filesystem on the device
(file:///private/var/mobile/Containers/Data/Application/A26B9C0B-0432-4D3F-91AF-880B3EBFD97D/tmp/flu
tter_application_1TWUWa1/flutter_application_1/)
[        ] Updating assets
[  +48 ms] Syncing files to device lxf的iPhone...
[   +1 ms] Compiling dart to kernel with 0 updated files
[        ] Processing bundle.
[        ] <- recompile package:flutter_application_1/main.dart f4c74373-4d9e-4ebc-bf2c-5e51ad43001f
[        ] <- f4c74373-4d9e-4ebc-bf2c-5e51ad43001f
[        ] Bundle processing done.
[  +29 ms] Updating files.
[        ] DevFS: Sync finished
[        ] Syncing files to device lxf的iPhone... (completed in 33ms)
[        ] Synced 0.0MB.
[        ] <- accept
[   +3 ms] Connected to _flutterView/0x105823c20.
[        ] Flutter run key commands.
[        ] r Hot reload. 🔥🔥🔥
[        ] R Hot restart.
[        ] h List all available interactive commands.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] 💪 Running with sound null safety 💪
[        ] An Observatory debugger and profiler on lxf的iPhone is available at:
           http://127.0.0.1:49444/Iv9OxkIC-sQ=/
[  +76 ms] The Flutter DevTools debugger and profiler on lxf的iPhone is available at:
           http://127.0.0.1:9104?uri=http://127.0.0.1:49444/Iv9OxkIC-sQ=/
```

### CODE REVIEW

**Comment by christopherfujino** [comment]
@jmagman WDYT about this change?

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/ios_deploy.dart:345]: Can you add a comment for why this check is needed, about the custom lldb prompt, etc?

**Review by jmagman — COMMENTED** [review]
This approach LGTM, minor nit about adding some comments.  Thank you for the contribution @LinXunFeng, I didn't think about the custom prompt case.

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/ios/ios_deploy.dart:345]: This LGTM once you add this comment

Comment (LinXunFeng) [reply packages/flutter_tools/lib/src/ios/ios_deploy.dart:347]: Thx, i have added the comment, please have a look.

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/ios_deploy.dart:293]:
```suggestion
  // This regex is to get the configurable lldb prompt. By default this prompt will be "lldb".
```

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/ios_deploy.dart:349]:
```suggestion
        // Use the configurable custom lldb prompt in the regex. The developer can set this prompt to anything. 
        // For example `settings set prompt "(mylldb)"` in ~/.lldbinit results in:
        // "(mylldb)    platform select remote-'ios' --sysroot"
```

**Review by jmagman — APPROVED** [review]
Minor suggestion for the comments. LGTM, and thanks again for the contribution!

**Review by jmagman — APPROVED** [review]
Still LGTM.

@christopherfujino when you get a chance can you add a second review?

**Review by christopherfujino — APPROVED** [review]
LGTM, thanks for the fix!

**Pull request merged**
